packages/workbox-range-requests/src/lib/handle-range-request.js

```diff
@@ -0,0 +1,188 @@
+ * @param {Object} input
+ * @param {Blob} input.blob A source blob.
+ * @param {Number|null} input.start The offset to use as the start of the
+ * slice.
+ * @param {Number|null} input.end The offset to use as the end of the slice.
+ * @return {Object} An object with `start` and `end` properties, reflecting
+ * the effective boundaries to use given the size of the blob.
+ */
+export function calculateEffectiveBoundaries({blob, start, end} = {}) {
+  isInstance({blob}, Blob);
+  const blobSize = blob.size;
+
+  if (end > blobSize || start < 0) {
+    throw ErrorFactory.createError('range-not-satisfiable');
+  }
+
+  let effectiveStart;
+  let effectiveEnd;
+
+  if (start === null) {
+    effectiveStart = blobSize - end;
+    effectiveEnd = blobSize;
+  } else if (end === null) {
+    effectiveStart = start;
+    effectiveEnd = blobSize;
+  } else {
+    effectiveStart = start;
+    // Range values are inclusive, so add 1 to the value.
+    effectiveEnd = end + 1;
+  }
+
+  return {
+    start: effectiveStart,
+    end: effectiveEnd,
+  };
+}
+
+/**
+ * Given a `Request` and `Response` objects as input, this will return a
+ * promise for a new `Response`.
+ *
+ * @param {Object} input
+ * @param {Request} input.request A request, which should contain a Range:
+ * header.
+ * @param {Response} input.response An original response containing the full
+ * content.
+ * @return {Promise<Response>} Either a `206 Partial Content` response, with
+ * the response body set to the slice of content specified by the request's
+ * `Range:` header, or a `416 Range Not Satisfiable` response if the
+ * conditions of the `Range:` header can't be met.
+ */
+export async function handleRangeRequest({request, response} = {}) {
+  try {
+    isInstance({request}, Request);
+    isInstance({response}, Response);
+
+    const rangeHeader = request.headers.get('range');
+    if (!rangeHeader) {
+      throw ErrorFactory.createError('no-range-header');
+    }
+
+    const boundaries = parseRangeHeader({rangeHeader});
+    const originalBlob = await response.blob();
+
+    const effectiveBoundaries = calculateEffectiveBoundaries({
+      blob: originalBlob,
+      start: boundaries.start,
+      end: boundaries.end,
+    });
+
+    const slicedBlob = originalBlob.slice(effectiveBoundaries.start,
+      effectiveBoundaries.end);
+    const slicedBlobSize = slicedBlob.size;
+
+    const slicedResponse = new Response(slicedBlob, {
+      // See https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/206
+      status: 206,
+      statusText: 'Partial Content',
+      headers: response.headers,
+    });
+
+    slicedResponse.headers.set('Content-Length', slicedBlobSize);
+    slicedResponse.headers.set('Content-Range',
+      `bytes ${effectiveBoundaries.start}-${effectiveBoundaries.end - 1}/` +
+      originalBlob.size);
+
+    return slicedResponse;
+  } catch (error) {
+    logHelper.warn({
+      message: `Unable to construct a sliced response; returning a 416 Range
```

Comment:
You should theoretically be able to get at it by examining the `request` that's logged in the `data.request` bit.